File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/db/TokenDao.java
package io.github.gusandrianos.foxforreddit.data.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import io.github.gusandrianos.foxforreddit.data.models.Token;

@Dao
public interface TokenDao {
    @Insert
    void insert(Token token);

    @Query("DELETE FROM token_table")
    void delete();

    @Query("SELECT * FROM token_table")
    List<Token> getToken();
}

File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/db/TokenDao.kt
package io.github.gusandrianos.foxforreddit.data.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import io.github.gusandrianos.foxforreddit.data.models.Token

@Dao
interface TokenDao {

    @Insert
    fun insert(token: Token)

    @Query("DELETE FROM token_table")
    fun delete()

    @get:Query("SELECT * FROM token_table")
    val token: List<Token>?
}
